tests/test_compile_kwargs.py

```diff
@@ -0,0 +1,555 @@
+from inspect import isclass
```

Comment:
I don't know if you want to import this during your tests. I don't think you want to encode the broad behavior that `isclass` encodes: `isclass` means "is this class instantiated?" `isclass(str) == True` but `isclass("fasdv") == False`.

I'd this code instead:

``` python
from tensorflow.keras.metrics import Metric
from tensorflow.keras.losses import Loss
from tensorflow.keras.optimizers import Optimizer

def test_optimizer_foo():
    ...
    if isinstance(optimizer, Optimizer):
        ...
```

Reply:
I agree. I've had issues in the past when working with TF doing checks like `isinstance(optimizer, Optimizer)` because they edit `sys.modules`, but in this case it works, so very much happy to replace!

Reply:
Ugh. That doesn't surprise me. Do you recall if those errors were on TF<2.0?

Reply:
To be honest I do not remember.